apis/appmesh/v1beta2/virtualnode_types.go

```diff
@@ -335,6 +338,9 @@ type DNSServiceDiscovery struct {
 	// +kubebuilder:validation:Enum=ENDPOINTS;LOADBALANCER
 	// +optional
 	ResponseType *string `json:"responseType,omitempty"`
+	// The ipPreference for the virtual node.
+	// +optional
+	IpPreference *string `json:"ipPreference,omitempty"`
```

Comment:
Add a kubebuilder annotation for allowed values for this field, the way we have for ResponseType

Reply:
Good call @cgchinmay

Reply:
fixed in new revision.